File: app/src/main/java/com/krystian/chessclock/timerPackage/Match.java
package com.krystian.chessclock.timerPackage;

import java.util.ArrayList;

public class Match {
    private Game currentGame;

    private int numberOfGames;
    private float firstPlayerPoints;
    private float secondPlayerPoints;
    private int gameNumber;

    private ArrayList<Integer> oneTimes;
    private ArrayList<Integer> oneIncrements;
    private ArrayList<Integer> twoTimes;
    private ArrayList<Integer> twoIncrements;

    public Match(int numberOfGames) {
        this.numberOfGames = numberOfGames;
        firstPlayerPoints = 0;
        secondPlayerPoints = 0;
        gameNumber = 1;
    }

    public Match() {
        this(1);
    }

    public Match(Match match) {
        this(match.numberOfGames, match.oneTimes, match.oneIncrements, match.twoTimes, match.twoIncrements);
    }

    public Match(int numberOfGames, ArrayList<Integer> oneTimes, ArrayList<Integer> oneIncrements,
                 ArrayList<Integer> twoTimes, ArrayList<Integer> twoIncrements) {
        this(numberOfGames);
        this.oneTimes = oneTimes;
        this.oneIncrements = oneIncrements;
        this.twoTimes = twoTimes;
        this.twoIncrements = twoIncrements;
    }

    public Match reset(Match match) {
        match.setGameNumber(1);
        match.setFirstPlayerPoints(0);
        match.setSecondPlayerPoints(0);
        match.setNumberOfGames(1);


        return match;
    }

    public Game getCurrentGame() { return currentGame; }
    public void setCurrentGame(Game currentGame) { this.currentGame = currentGame; }

    public int getNumberOfGames() { return numberOfGames; }
    public void setNumberOfGames(int numberOfGames) { this.numberOfGames = numberOfGames; }

    public float getFirstPlayerPoints() { return firstPlayerPoints; }
    public void setFirstPlayerPoints(float firstPlayerPoints) { this.firstPlayerPoints = firstPlayerPoints; }

    public float getSecondPlayerPoints() { return secondPlayerPoints; }
    public void setSecondPlayerPoints(float secondPlayerPoints) { this.secondPlayerPoints = secondPlayerPoints; }

    public int getGameNumber() { return gameNumber; }
    public void setGameNumber(int gameNumber) { this.gameNumber = gameNumber; }

    public ArrayList<Integer> getOneTimes() { return oneTimes; }
    public void setOneTimes(ArrayList<Integer> oneTimes) { this.oneTimes = oneTimes; }

    public ArrayList<Integer> getOneIncrements() { return oneIncrements; }
    public void setOneIncrements(ArrayList<Integer> oneIncrements) { this.oneIncrements = oneIncrements; }

    public ArrayList<Integer> getTwoTimes() { return twoTimes; }
    public void setTwoTimes(ArrayList<Integer> twoTimes) { this.twoTimes = twoTimes; }

    public ArrayList<Integer> getTwoIncrements() { return twoIncrements; }
    public void setTwoIncrements(ArrayList<Integer> twoIncrements) { this.twoIncrements = twoIncrements; }
}

File: app/src/main/java/com/krystian/chessclock/timerPackage/Match.kt
package com.krystian.chessclock.timerPackage

class Match @JvmOverloads constructor(@JvmField var numberOfGames: Int = 1) {
    @JvmField
    var currentGame: Game? = null
    @JvmField
    var firstPlayerPoints = 0f
    @JvmField
    var secondPlayerPoints = 0f
    @JvmField
    var gameNumber = 1
    lateinit var oneTimes: ArrayList<Int?>
    lateinit var oneIncrements: ArrayList<Int?>
    lateinit var twoTimes: ArrayList<Int?>
    lateinit var twoIncrements: ArrayList<Int?>

    constructor(match: Match) : this(
        match.numberOfGames,
        match.oneTimes,
        match.oneIncrements,
        match.twoTimes,
        match.twoIncrements
    )

    constructor(
        numberOfGames: Int, oneTimes: ArrayList<Int?>, oneIncrements: ArrayList<Int?>,
        twoTimes: ArrayList<Int?>, twoIncrements: ArrayList<Int?>
    ) : this(numberOfGames) {
        this.oneTimes = oneTimes
        this.oneIncrements = oneIncrements
        this.twoTimes = twoTimes
        this.twoIncrements = twoIncrements
    }

    fun reset(match: Match): Match {
        match.gameNumber = 1
        match.firstPlayerPoints = 0f
        match.secondPlayerPoints = 0f
        match.numberOfGames = 1
        return match
    }
}
